src/main/kotlin/com/amazon/opendistroforelasticsearch/indexmanagement/indexstatemanagement/util/ManagedIndexUtils.kt

```diff
@@ -86,9 +89,22 @@ fun managedIndexConfigIndexRequest(managedIndexConfig: ManagedIndexConfig): Inde
             .id(managedIndexConfig.indexUuid)
             .setIfPrimaryTerm(managedIndexConfig.primaryTerm)
             .setIfSeqNo(managedIndexConfig.seqNo)
+            .routing(managedIndexConfig.indexUuid) // we want job doc and its metadata doc be routed to same shard
             .source(managedIndexConfig.toXContent(XContentFactory.jsonBuilder(), ToXContent.EMPTY_PARAMS))
 }
 
+fun managedIndexMetadataIndexRequest(managedIndexMetadata: ManagedIndexMetaData): IndexRequest {
+    // routing set using managed index's uuid
+    // so that metadata doc and managed-index doc are in the same place
+    return IndexRequest(INDEX_MANAGEMENT_INDEX)
+            .id(managedIndexMetadata.indexUuid + "metadata")
```

Comment:
Can you add a # delimiter between indexUuid and metadata. Also I see it used below too, can you turn it into a helper function, i.e. (uuid) -> "$uuid#metadata" type thing so you only have it defined in one place.

Reply:
good idea, changed